src/main/java/lotto/domain/Winners.java

```diff
@@ -6,33 +6,35 @@ public class Winners {
     public static final int COUNT_UNIT = 1;
     public static final int COUNT_INITIAL = 0;
 
-    private final Map<WinningsType, Integer> winners;
+    private final Map<Rank, Integer> winners;
 
-    public Winners() {
+    private Winners() {
         this.winners = new LinkedHashMap<>();
-        for (WinningsType winningsType : WinningsType.values()) {
+        for (Rank winningsType : Rank.values()) {
             winners.put(winningsType, COUNT_INITIAL);
         }
     }
 
-    public Map<WinningsType, Integer> getWinners() {
-        return new LinkedHashMap<>(winners);
+    public static Winners of() {
+        return new Winners();
     }
 
-    public void findWinners(Lotto lotto, WinningNumbers winningNumbers) {
-        int count = winningNumbers.numberOfSame(lotto);
-        WinningsType temporary = WinningsType.selectWinningsType(count, winningNumbers.isSameBonus(lotto));
-        addWinner(temporary);
+    public Map<Rank, Integer> getWinners() {
+        return new LinkedHashMap<>(winners);
     }
 
-    public void addWinner(WinningsType winningsType) {
-        winners.put(winningsType, winners.get(winningsType) + COUNT_UNIT);
+    public void findWinners(Lottos lottos, WinningNumbers winningNumbers) {
+        List<Rank> ranks = lottos.selectRankType(winningNumbers);
+        ranks.forEach(rank -> addWinner(rank));
```

Comment:
```suggestion
        ranks.forEach(this::addWinner);
```

저는 개인적으로 좀 더 간단하게 표현할 수 있어서 메서드 레퍼런스 형태를 선호하긴 합니다 😄

Reply:
참고하겠습니다..